src/rules/native-attribute-binding.ts

```diff
@@ -0,0 +1,140 @@
+
+import * as expressions from '../expressions';
+import * as matchers from '../matchers';
+import {Rule} from '../rule';
+
+export const a11yAttributes: Set<string> = new Set([
+  'aria-activedescendant',
+  'aria-atomic',
+  'aria-autocomplete',
+  'aria-busy',
+  'aria-checked',
+  'aria-controls',
+  'aria-describedby',
+  'aria-disabled',
+  'aria-dropeffect',
+  'aria-expanded',
+  'aria-flowto',
+  'aria-grabbed',
+  'aria-haspopup',
+  'aria-hidden',
+  'aria-invalid',
+  'aria-label',
+  'aria-labelledby',
+  'aria-level',
+  'aria-live',
+  'aria-multiline',
+  'aria-multiselectable',
+  'aria-orientation',
+  'aria-owns',
+  'aria-posinset',
+  'aria-pressed',
+  'aria-readonly',
+  'aria-relevant',
+  'aria-required',
+  'aria-selected',
+  'aria-setsize',
+  'aria-sort',
+  'aria-valuemax',
+  'aria-valuemin',
+  'aria-valuenow',
+  'aria-valuetext',
+  'role'
+]);
+
+export const nativeAttributes: Set<string> = new Set([
+  'accesskey',
+  'class',
+  'contenteditable',
+  'contextmenu',
+  'dir',
+  'draggable',
+  'dropzone',
+  'hidden',
+  'href',
+  'id',
+  'itemprop',
+  'lang',
+  'spellcheck',
+  'style',
+  'style',
+  'tabindex',
+  'title'
+]);
+
+const expressionParser = new expressions.ExpressionParser();
+
+export class NativeAttributeBinding implements Rule {
+  public async check(document: Document): Promise<Warning[]> {
+    const warnings: Warning[] = [];
+    const templates = dom5.queryAll(
+        document.parsedDocument.ast,
+        dom5.predicates.OR(
+            matchers.isDomBindTemplate, matchers.isDomModuleTemplate));
+
+    for (const template of templates) {
+      for (const element of dom5.queryAll(
+               parse5.treeAdapters.default.getTemplateContent(template),
+               (e) => true)) {
+        for (const attr of element.attrs) {
+          if (this._isBindingExpression(attr.value) &&
+              !this._canAttributeUseNativeBinding(element, attr)) {
+            const parsedHtml = document.parsedDocument;
+            if (!(parsedHtml instanceof ParsedHtmlDocument)) {
+              continue;
+            }
+            warnings.push({
+              code: 'native-attribute-binding',
+              message: `The expression ${attr.value} bound to attribute ` +
+                  `'${attr.name}' should use $= instead of =`,
+              severity: Severity.ERROR,
+              sourceRange:
+                  parsedHtml.sourceRangeForAttribute(element, attr.name)!
+            });
+          }
+        }
+      }
+    }
+    return warnings;
+  }
+
+  private _canAttributeUseNativeBinding(
+      element: parse5.ASTNode,
+      attr: parse5.ASTAttribute): boolean {
+    const name: string = attr.name.toLowerCase();
+    if (name === 'for' && dom5.predicates.hasTagName('label')(element)) {
+      return false;
+    }
+    if (nativeAttributes.has(name)) {
+      return false;
+    }
+    if (name.indexOf('data-') === 0 && attr.name[name.length - 1] !== '$') {
```

Comment:
Why are we checking whether attr.name ends with `$` here? Shouldn't it be checked elsewhere already?